### PR TITLE
Gitops serverless

### PR DESCRIPTION
**Pull request by m00g3n**
**Description**

Changes proposed in this pull request:

- Gitops support for functions

**Related issue(s)**

See also: https://github.com/kyma-project/kyma/issues/5752
See also: https://github.com/kyma-project/kyma/issues/8959

### CODE REVIEW

File: components/function-controller/internal/controllers/serverless/build_resources.go [hunk 1]
```diff
@@ -17,6 +17,8 @@ import (
 const (
 	destinationArg        = "--destination"
 	functionContainerName = "lambda"
+	//TODO refactor strings with workspace path
```
Comment (m00g3n) [inline components/function-controller/internal/controllers/serverless/build_resources.go:20]: refactor will be applied in https://github.com/kyma-project/kyma/pull/9111/files

File: components/function-controller/internal/controllers/serverless/function_controller.go [hunk 2]
```diff
@@ -64,15 +64,34 @@ func (r *FunctionReconciler) calculateImageTag(instance *serverlessv1alpha1.Func
 }
 
 func (r *FunctionReconciler) updateStatus(ctx context.Context, result ctrl.Result, instance *serverlessv1alpha1.Function, condition serverlessv1alpha1.Condition) (ctrl.Result, error) {
+	return r.updateStatus2(ctx, result, instance, condition, nil)
```
Comment (m00g3n) [inline components/function-controller/internal/controllers/serverless/function_controller.go:67]: refactor will be applied in https://github.com/kyma-project/kyma/pull/9111/files

File: components/function-controller/internal/controllers/serverless/function_reconcile.go [hunk 3]
```diff
@@ -120,3 +148,59 @@ func (r *FunctionReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
...
+	return r.updateStatus2(ctx, ctrl.Result{}, instance, serverlessv1alpha1.Condition{
+		Type:               serverlessv1alpha1.ConditionConfigurationReady,
+		Status:             corev1.ConditionTrue,
+		LastTransitionTime: metav1.Now(),
+		Reason:             serverlessv1alpha1.ConditionReasonSourceUpdated,
+		Message:            fmt.Sprintf("Sources %s updated", instance.Name),
+	}, repository)
+}
+
+var gitOperator = git.New()
```
Comment (aerfio) [inline components/function-controller/internal/controllers/serverless/function_reconcile.go:170]: gitOperator is meant to be a global? or do you want to refactor it with the rest, in next PR?

File: components/function-controller/internal/controllers/serverless/build_resources.go [hunk 4]
```diff
@@ -132,6 +134,150 @@ func (r *FunctionReconciler) buildJob(instance *serverlessv1alpha1.Function, con
...
+									corev1.ResourceMemory: r.config.Build.RequestsMemoryValue,
+									corev1.ResourceCPU:    r.config.Build.RequestsCPUValue,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								// Must be mounted with SubPath otherwise files are symlinks and it is not possible to use COPY in Dockerfile
+								// If COPY is not used, then the cache will not work
+								{Name: "workspace", MountPath: workspace},
+								{Name: "credentials", ReadOnly: true, MountPath: "/docker"},
+								{Name: "runtime", ReadOnly: true, MountPath: "/workspace/Dockerfile", SubPath: "Dockerfile"},
```
Comment (aerfio) [inline components/function-controller/internal/controllers/serverless/build_resources.go:265]: update that comment, if it's valid only for dockerfile in this case please move it 2 lines down

File: components/function-controller/internal/controllers/serverless/function_reconcile.go (hunk 3 above)
Comment (m00g3n) [reply components/function-controller/internal/controllers/serverless/function_reconcile.go:170]: It will be fixed in upcoming PR, thx.

**Pull request merged**